dlt/common/storages/fsspec_filesystem.py

```diff
@@ -252,6 +296,32 @@ def guess_mime_type(file_name: str) -> Sequence[str]:
     return type_
 
 
+def extract_mtime(file_metadata: Dict[str, Any], protocol: str = None) -> pendulum.DateTime:
+    """Extract the modification time from file listing metadata.
+
+    Args:
+        file_metadata (Dict[str, Any]): The file metadata.
+        protocol (str) [Optional]: The protocol. If not provided, None or not a known protocol,
+            then default field name `mtime` is tried. `mtime` is used for the "file" fsspec
+            implementation and our custom fsspec implementations.
+
+    Returns:
+        pendulum.DateTime: The modification time.
+
+    Raises:
+        KeyError: If the resolved field name is not found in the metadata. Current dlt use-cases
+            depend on a modified date. For example, transactional files, incremental destination
+            loading.
+    """
+    field_name = MTIME_FIELD_NAMES.get(protocol, DEFAULT_MTIME_FIELD_NAME)
+    try:
+        return ensure_pendulum_datetime(file_metadata[field_name])
+    except KeyError:
+        if protocol not in MTIME_FIELD_NAMES:
```

Comment:
It's good enough.   I set to `pendulum.now()`      ...for now  🤣 ...  

Always Hard to tell with these unknowns 😓.   `now()` duplicates `loaded_at` etc, which pipelines already have.   And it could be misleading.

It would be better to use something that indicates that no info was available:  
- `None`.    will it break the pipeline downstream?
- Jan 1st 1970.   ie, epoch zero.    It's ugly but people know what it means.